### DISCUSSION TITLE
Displaying MessageID via Powershell

### DISCUSSION THREAD

**Discussion by Jochonon** [community/community#123009] [Programming Help]
### Body

Hi guys.
Please, tell me if there is an opportunity to run searching and displaying an MessageID of every single email, which I will get via my code.
I was trying to figure out it, but I have only blank column without any messageid record.
My code is located in attachment. Many thanks for your answers.
[EmailTracking.txt](https://github.com/community/community/files/15252829/EmailTracking.txt)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by kazmatw** [comment]
If you have access to Exchange Management Shell or can use Exchange Online PowerShell, you can use the Get-MessageTrackingLog cmdlet, which is designed for message tracking. Here’s how you might structure your command:

```powershell
# Ensure you're connected to Exchange PowerShell
Get-MessageTrackingLog -ResultSize Unlimited | Select-Object Timestamp, Sender, Recipients, MessageSubject, MessageId | Format-Table
```
This command retrieves all messages from the message tracking logs and displays the timestamp, sender, recipients, subject, and MessageId. Adjust the ResultSize and other parameters based on your need.
Good luck!
